library/alloc/src/vec/splice.rs

```diff
@@ -54,6 +54,11 @@ impl<I: Iterator, A: Allocator> ExactSizeIterator for Splice<'_, I, A> {}
 impl<I: Iterator, A: Allocator> Drop for Splice<'_, I, A> {
     fn drop(&mut self) {
         self.drain.by_ref().for_each(drop);
+        // At this point draining is done and the only remaining tasks are splicing
+        // and moving things into the final place.
+        // Which means we can replace the slice::Iter with pointers that won't point to deallocated
+        // memory. This makes Miri happy.
```

Comment:
It's not just about pleasing Miri, right?

```suggestion
        // memory, and `Drain::drop` can still see that it's empty. This also makes Miri happy.
```

Reply:
Well, logically it already was 0-length before, just the way it calculated the way was Miri-UB but not LLVM-UB because we're not emitting any hints that llvm could exploit. But that rephrase is not quite right either, it's about Drain being allowed to calculate the length at all.

Reply:
Updated it.